internal/controllers/clusterrequest_controller.go

```diff
@@ -705,6 +715,80 @@ func (t *clusterRequestReconcilerTask) handleClusterInstallation(ctx context.Con
 	return nil
 }
 
+func (t *clusterRequestReconcilerTask) handleClusterPolicyConfiguration(ctx context.Context) error {
+	// Get all the policies.
+	policies := &policiesv1.PolicyList{}
+
+	err := t.client.List(ctx, policies)
```

Comment:
We can't scope it to clusters unfortunately since the clusters and the associated compliance are included in the `status.status` of the policy which is of array type and cannot be filtered directly in the controller-runtime `List` function.
We could, however, get just those policies in the namespaces that are `ztp` + <cluster request namespace>.
It  should be clearly explained in the documentation that policies should only be created in those namespaces though.

Reply:
Even if we scoped this to just "ztp-" namespace policies (which probably isn't the best solution), there are still going to be 1000+ policies on a large O-Cloud. And if we have 100 config change operations in progress and each operation is being reconciled in its own thread (correct me if I'm wrong on that) it will result in a lot of k8s API operations with large amounts of data being returned. Does TALM have the same issue in large clusters? How does it deal with it?

Reply:
I should mention that I don't think we need to address this now if there isn't an easy way to do it. If you do think this could be a scaling issue down the road though, might be worth adding a comment.

Reply:
Regarding the concern about the List query, the List query on a typed object is from the cache, so it should only take milliseconds, even if there are thousands of policies.

Reply:
After further discussion with @Missxiaoguo , we agreed it would be better for this usecase to query for the child policies directly instead of parent policies. I will make the updates today.
Thank you!

Reply:
The new version looks good.